marathon/logging.go

```diff
@@ -0,0 +1,97 @@
+package marathon
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"gopkg.in/clever/kayvee-go.v2"
+)
+
+type m map[string]interface{}
+
+func LogState(apps []App, logMarathonTasks bool) {
+	var summary struct {
+		appCount     int64
+		runningCount int64
+		stagedCount  int64
+	}
+	for _, app := range apps {
+		summary.appCount++
+		summary.runningCount += app.TasksRunning
+		summary.stagedCount += app.TasksStaged
+		logApp(app, logMarathonTasks)
+	}
+	summaryLog := m{
+		"source":       "marathon-stats",
+		"title":        "marathon-summary",
+		"totalApps":    summary.appCount,
+		"runningTasks": summary.runningCount,
+		"stagedTasks":  summary.stagedCount,
+		"type":         "gauge",
+	}
+	log.Println(kayvee.Format(summaryLog))
+}
+
+func logApp(app App, logMarathonTasks bool) {
+	normalizedAppID := normalizeAppID(app.ID)
+	appLog := m{
+		"source":         "marathon-stats",
+		"title":          fmt.Sprintf("marathon-apps.%s", normalizedAppID),
+		"level":          kayvee.Info,
+		"appId":          app.ID,
+		"instances":      app.Instances,
+		"cpus":           app.Cpus,
+		"mem":            app.Mem,
+		"disk":           app.Disk,
+		"image":          app.Container.Docker.Image,
+		"version":        app.Version,
+		"tasksStaged":    app.TasksStaged,
+		"tasksHealthy":   app.TasksHealthy,
+		"tasksRunning":   app.TasksRunning,
+		"tasksUnhealthy": app.TasksUnhealthy,
+		"type":           "gauge", // This is to auto load metric into influx
+	}
```

Comment:
By emitting all the stats in a single block, I believe that makes it a bit harder to auto-handle from the Heka side. This is at least true for SignalFx; I'm not 100% about Influx -- can it write multiple values per stat?

If not, We would need to write Heka filters for each one of these, specify which field to pull the value from, and give it a new name so it fits the `<source>.<title>` convention.

What if we instead emitted one stat per line, e.g. `marathon-stats.cpus`, `marathon-stats.mem`, etc... where each one has a `type` and `value` (relevant [infra-docs](https://github.com/Clever/infra-docs/blob/master/monitor/application-metrics.md#example))